assets/js/components/setup/compatibility-checks.js

```diff
@@ -65,6 +66,31 @@ const checks = [
 			throw ERROR_TOKEN_MISMATCH;
 		}
 	},
+	// Check that server can connect to Google's APIs via the core/site/data/health-checks endpoint.
+	async () => {
+		try {
+			const response = await API.get( 'core', 'site', 'health-checks', undefined, {
+				useCache: false,
+			} );
+
+			if ( ! response?.checks?.googleAPI?.pass ) {
+				throw ERROR_GOOGLE_API_CONNECTION_FAIL;
+			}
+		} catch ( error ) {
+			throw ERROR_GOOGLE_API_CONNECTION_FAIL;
+		}
```

Comment:
If the rest request fails to the `health-checks` endpoint, it should not throw the `ERROR_GOOGLE_API_CONNECTION_FAIL` error. Instead this should throw `ERROR_FETCH_FAIL` - see above.